### PR TITLE
Track and report test coverage

### PR DESCRIPTION
**Pull request by nylen**
Use Jest code coverage added in #1382 (with some tweaks/improvements in this PR to ignore built files, Storybook stories, etc.) and [coveralls.io](https://coveralls.io/github/WordPress/gutenberg) and [`node-coveralls`](https://github.com/nickmerwin/node-coveralls) for reporting coverage during Travis CI builds.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
-gutenberg.zip
+node_modules
 storybook-static
+
+# Files that may be generated by this project
+gutenberg.zip
+
+# Files that may appear in your environment
 .DS_Store
+.vscode
```
Comment (ntwb) [inline .gitignore]: I see `.vscode` was already here and this is just moving its location, though should it really be here in the first place?

Via https://github.com/WordPress/gutenberg/pull/1382#discussion_r125820857 from a couple of days ago:
> IDE specific entries in `.gitignore` should not be in the repo, users should manage these locally in `~/.gitignore_global`
>
> See also https://core.trac.wordpress.org/ticket/34345

**Comment by ntwb** [comment]
Coveralls comments and notifications can get noisy, I think turning off comments and changing the _failure_ threshold to 5% would be something we should do:

Via https://coveralls.io/github/WordPress/gutenberg/settings
• Uncheck the _"LEAVE COMMENTS?"_ checkbox
• Change the _"COVERAGE DECREASE THRESHOLD FOR FAILURE"_ value to 5%

**Review by ntwb — APPROVED** [review]
LGTM 👍

**Comment by nylen** [comment]
> Coveralls comments and notifications can get noisy, I think turning off comments and changing the failure threshold to 5% would be something we should do

Maybe so.  However, coverage is at 18%.  **Eighteen percent.**  This shows that we are not developing with testing in mind.

So I'd like to leave this enabled for a while.

File: .gitignore (hunk 1 above)
Comment (nylen) [inline .gitignore]: :+1: Originally added in #294.

**Comment by aduth** [comment]
With these changes, local testing and Travis testing don't produce the same result. You can see that running `npm test` locally fails on the master branch with an error "Your test suite must contain at least one test" (because it's trying to run e.g. `test/setup-globals.js` as a test, fix pending). I see that `npm run test-unit:coverage-ci` runs the test, but does not fail with the same error.

**Comment by aduth** [comment]
Hmm, actually, `npm run test-unit:coverage-ci` _does_ fail with the same error for me. Not sure why yet...

**Pull request merged**